src/benchcab/comparison.py

```diff
@@ -38,11 +38,26 @@ def __init__(
         self.task_name = task_name
         self.logger = get_logger()
 
+    def _get_state_dir(self) -> Path:
```

Comment:
See below, perhaps create a file of functions in the utils folder called state.py, which contains all of the state-related methods that take the different usages as arguments.